### ISSUE TITLE
Tasks Applicability Study Settings / Travel Mode Detection 🚎

### ISSUE THREAD

**Issue by fuzsh** [Yasoz/UniTraj#3]
Hi,

Nice work. I had a question regarding the Tasks Applicability section (page 28-29 -- [Paper](https://openreview.net/pdf/a1a6a045a43be34e5e45854f86a202ecc87cfc17.pdf)).

Could you clarify whether the code for the following tasks will be made available? 

- Trajectory Prediction
- Trajectory Classification

I'm curios about the section of the adaptability. If these components are already released, could you point me to them?

Thanks!

**Comment by Yasoz** [comment]
Thank you for your interest. For the trajectory prediction task, you only need to adjust the position of the mask to the last points to be predicted. 
For trajectory classification, the following code is for your reference:

```python
pretrained_model = UniTraj(
        trajectory_length=200,
        patch_size=1,
        embedding_dim=128,
        encoder_layers=8,
        encoder_heads=4,
        decoder_layers=4,
        decoder_heads=4,
        mask_ratio=0.0,
    )
    
class TrajectoryClassifier(nn.Module):
    def __init__(self, pretrained_uni_traj, num_classes):
        super(TrajectoryClassifier, self).__init__()
        self.encoder = pretrained_uni_traj.encoder
        self.interval_embedding = pretrained_uni_traj.interval_embedding
        self.embedding_dim = 128
        self.classifier = nn.Sequential(
            nn.BatchNorm1d(self.embedding_dim),
            nn.Linear(self.embedding_dim, num_classes)
        )
    
    def forward(self, trajectory, intervals=None, mask_indices=None):
        B, _, L = trajectory.shape
        # calculate interval embeddings
        if intervals is not None:
            intervals = intervals.unsqueeze(-1)  # [B, L, 1]
            interval_embeddings = self.interval_embedding(intervals)  # [B, L, embedding_dim]
        else:
            # if no intervals are provided, use zero vectors
            interval_embeddings = torch.zeros(B,L, self.embedding_dim, device=trajectory.device)
        # extract features using the encoder
        features, _ = self.encoder(trajectory, interval_embeddings, mask_indices)
        cls_token = features[0]  # [B, C]
        logits = self.classifier(cls_token)  # [B, num_classes]
        return logits

```
